py3status/modules/google_calendar.py

```diff
@@ -0,0 +1,438 @@
+    2. OAuth 2.0 credentials for the Google Calendar api.
+
+    Follow Step 1 of the guide here to obtain your OAuth 2.0 credentials:
+    https://developers.google.com/google-apps/calendar/quickstart/python
+
+    Download the client_secret.json file which contains your client ID and
+    client secret. In your i3status config file, set configuration parameter
+    client_secret_path to the path to your client_secret.json file.
+
+    The first time you run the module, a browser window will open asking you
+    to authorize access to your calendar. After authorization is complete,
+    an access/refresh token will be saved to the path configured in
+    auth_token_path, and i3status will be restarted. This restart will
+    occur only once after the first time you successfully authorize.
+
+Example:
+```
+google_calendar {
+    client_secret_path = '/path/to/client_secret/'
+    auth_token_path = '/path/to/auth_token/'
+}
+```
+@author Igor Grebenkov
+@license BSD
+"""
+
+import httplib2
+import os
+import datetime
+import webbrowser
+
+from apiclient import discovery
+from oauth2client import client
+from oauth2client import tools
+from oauth2client.file import Storage
+from httplib2 import ServerNotFoundError
+
+
+SCOPES = 'https://www.googleapis.com/auth/calendar.readonly'
+APPLICATION_NAME = 'py3status google_calendar module'
+
+
+def gstr_to_datetime(date_time_str):
+    """ Returns a datetime object from a google calendar date/time string."""
+    tmp = '-'.join(date_time_str.split('-')[:-1])
+    return datetime.datetime.strptime(tmp, '%Y-%m-%dT%H:%M:%S')
+
+
+def datetime_to_str(date_time, dt_format):
+    """ Returns a strftime formatted string from a datetime object."""
+    return date_time.strftime(dt_format)
+
+
+def delta_time(date_time):
+    """
+    Returns in a dict the number of days/hours/minutes and total minutes
+    until date_time.
+    """
+    now = datetime.datetime.now()
+    diff = date_time - now
+
+    days = int(diff.days)
+    hours = int(diff.seconds / 3600)
+    minutes = int((diff.seconds / 60) - (hours * 60)) + 1
+    total_minutes = int((diff.seconds / 60) + (days * 24 * 60)) + 1
+
+    return {'days': days,
+            'hours': hours,
+            'minutes': minutes,
+            'total_minutes': total_minutes}
+
+
+class Py3status:
+    auth_token_path = '~/.credentials'
+    button_open_url = 3
+    button_toggle_format = 1
+    cache_timeout = 60
+    client_secret_path = ''
+    colors = []
+    event_in_progress_suffix = 'Remaining'
+    event_separator = '|'
+    format = '{events}'
+    format_date = '%a %d-%m'
+    format_error = ''
+    format_event_compact = '{summary} {time_until}'
+    format_event_full = '{summary} ({start_time} - {end_time}, {start_date})'
+    format_time = '%I:%M %p'
+    format_time_until = '({days}d {hours}h {mins}m)'
+    num_events = 3
+    time_until_threshold = 180
+    warn_threshold = 0
+    warn_timeout = 300
+
```

Comment:
(Opinion) Some config names may be too specific / long. See if we can simplify the settings to be more nice for the users to specify/keep. `auth_token`, `button_open`, `button_toggle`, `client_secret`, etc.

Reply:
Shortened/simplified config parameters.

Reply:
(Opinion). I like the old `format_time_until` better over the new `format_time_to`. I'm thinking maybe `format_timer` would look nice in the long run. Will try and test this module when I can.